### PR TITLE
[FEAT][BWS]: endpoint to get rates of all coins

### PR DESCRIPTION
**Pull request by rastajpa**

### CODE REVIEW

File: packages/bitcore-wallet-service/src/lib/expressapp.ts [hunk 1]
```diff
@@ -1179,7 +1197,7 @@ export class ExpressApp {
       } catch (ex) {
         return returnError(ex, res, req);
       }
-      server.getFiatRates(opts, (err, rates) => {
+      server.getFiatRatesByCoin(opts, (err, rates) => {
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/expressapp.ts:1200]: nice.

File: packages/bitcore-wallet-service/src/lib/fiatrateservice.ts [hunk 2]
```diff
@@ -168,6 +153,56 @@ export class FiatRateService {
...
+      },
+      (err, res: any) => {
+        if (err) return cb(err);
+        return cb(null, Object.assign({}, ...res));
+      }
+    );
+  }
+
+  getRatesByCoin(opts, cb) {
+    $.shouldBeFunction(cb, 'Failed state: type error (cb not a function) at <getRatesByCoin()>');
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/fiatrateservice.ts:204]: maybe add:

    if (!checkRequired(opts, ['coin'], cb)) return;
(so, coin is required always).

File: packages/bitcore-wallet-service/src/lib/fiatrateservice.ts [hunk 3]
```diff
@@ -168,6 +153,56 @@ export class FiatRateService {
...
     $.shouldBeFunction(cb, 'Failed state: type error (cb not a function) at <getRates()>');
 
+    opts = opts || {};
+
+    const now = Date.now();
+    const ts = opts.ts ? opts.ts : now;
+    let fiatFiltered = [];
+    let rates = [];
+
+    if (opts.code) {
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/fiatrateservice.ts:163]: is code really optional? what is the expected behavior is not given?

**Review by matiu — CHANGES_REQUESTED** [review]
great job! Left some minor comments.

File: packages/bitcore-wallet-service/src/lib/fiatrateservice.ts (hunk 2 above)
Comment (rastajpa) [reply packages/bitcore-wallet-service/src/lib/fiatrateservice.ts:204]: this check is done in the previous step. Check server.ts

File: packages/bitcore-wallet-service/src/lib/fiatrateservice.ts (hunk 3 above)
Comment (rastajpa) [reply packages/bitcore-wallet-service/src/lib/fiatrateservice.ts:163]: Yes, it is optional, if you don't pass it as a parameter then you get the rates in all fiat supported. if you pass it as a parameter you only get the rate of the specified fiat (code)

**Pull request merged**